test/pythonml/test_workflow_integration.py

```diff
@@ -0,0 +1,175 @@
+import unittest
+import os, shutil, sys
+import re
+import subprocess
+import functools
+import yaml
+from parameterized import parameterized
+
+with open("integration_configuration.yaml", "r") as inp:
+    configuration = yaml.safe_load(inp)
+
+unit_shortnames = configuration["unit_shortnames"]
+
+# ToDo: Refactor data clumps below into classes
+
+# Figure out paths
+asset_path = configuration["asset_path"]
+fixtures_path = configuration["fixtures"]["path"]
+settings_filename = configuration["fixtures"]["settings"]
+
+
+# Get the path to the training data
+def get_dataset_filenames(config):
+    training_set_name = config["training_set_name"]
+    predict_set_name = config["predict_set_name"]
+    return training_set_name, predict_set_name
+
+
+regression_training_file, regression_predict_file = get_dataset_filenames(
+    configuration["fixtures"]["regression"])
+classification_training_file, classification_predict_file = get_dataset_filenames(
+    configuration["fixtures"]["classification"])
+clustering_training_file, clustering_predict_file = get_dataset_filenames(
+    configuration["fixtures"]["clustering"])
+
+# Which files should we remove
+files_to_remove = configuration["files_to_remove"]
+extensions_to_remove = configuration["extensions_to_remove"]
+
+
+# Extract the list of tests
+def get_test_names_configs(configuration):
+    names = [i[0] for i in configuration]
+    tests = [i[1]["units_to_run"] for i in configuration]
+    return tests, names
+
+
+all_tests = configuration["tests"]
+
+regression_configs = list(filter(lambda i: i[1]["category"] == "regression", all_tests.items()))
```

Comment:
I think the preferred way to do this sort of behavior in python is with a list comprehension:
```
regression_configs = [x for x in all_tests.items() if x[1]["category"] == "regression"]
```
It's equivalent, and it's fine the way it is, but lambdas seem to have a bad reputation in the python world.
Guido argued, during the python 3 development era, to remove them:
https://www.artima.com/weblogs/viewpost.jsp?thread=98196

Reply:
I'm a fan of lambdas and functional stuff, but I agree that the list comprehension makes this a bit more readable - adjusted this accordingly